src/Test/Utilities/Shared/Syntax/SourceUtilities.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Roslyn.Test.Utilities.Syntax
+{
+    internal sealed class RandomizedSourceText: SourceText
+    {
+        private char hash(int position)
+        {
+            // good enough "random looking" hasher via the xorshift128plus algorithm.
+            unchecked
+            {
+                UInt64 s1 = (UInt64) ~position;
+                UInt64 s0 = (UInt64) position;
+                s1 ^= s1 << 23;
+                s1 ^= s1 >> 17;
+                s1 ^= s0;
+                s1 ^= s0 >> 26;
+                return (char)(s1 + s0);
+            }
+        }
+
+        public override char this[int position] => hash(position);
+
+        public override Encoding Encoding => Encoding.UTF8;
+
+        public override int Length
+        {
+            get
+            {
+                return 40 * 1000 * 1000;
+            }
+        }
+
+        public override void CopyTo(int sourceIndex, char[] destination, int destinationIndex, int count)
+        {
+            throw new UnauthorizedAccessException();
```

Comment:
Oh, I threw that in there to see if `CopyTo` was being used.  Turns out it was (as you can see in the test failures).

When @gafter asked me to push my fixes to other issues, I forgot to change this back to it's actual implementation.